### ISSUE TITLE
Got Nan of loss value when training

### ISSUE THREAD

**Issue by youjinqing123** [open-mmlab/mmdetection#418]
We are training our own dataset with retinanet 50,we changed the data annotation into pickle just like illustrated in custom.py,  the data type of bbox and labels is int32, sicnce we don't have ignore images,we set it like this ,e.g  ''a.jpg'' has two objects,
{
        'filename': 'a.jpg',
        'width': 1280,
        'height': 720,
        'ann': {
            'bboxes': np.array( [[xmin_1(int32), ymin_1,  width_1,   height_1],
                                          [xmin_2, ymin_2,  width_2,   height_2]]),
            'labels': np.array([label1(int32),label2]),
        }
}
After we set config and ran the train.sh,we met some problem like expected Long but Int is given,
expected DoubleTensor but FloatTensor is given,we have to add some codes like .double() to slove these issues.We didn't change anything else,just make sure the type of data is right.We use pytorch 1.0 ,should we do things like .double(),or we just didn't handle the data well?

Finally it started to run,but we found the value of clf loss,reg loss were nan,could you tell us the reasons may cause it?We have 5 classes and we have already changed lr, cause we just use one gpu, here is our config file

# model settings
model = dict(
    type='RetinaNet',
    #pretrained='modelzoo://resnet50',
    pretrained='/home/ubuntu/MyFiles/mmdetection/models/resnet50-19c8e357.pth',
    #pretrained=None,
    backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=1,
        style='pytorch'),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        start_level=1,
        add_extra_convs=True,
        num_outs=5),
    bbox_head=dict(
        type='RetinaHead',
        num_classes=6,
        in_channels=256,
        stacked_convs=4,
        feat_channels=256,
        octave_base_scale=4,
        scales_per_octave=3,
        anchor_ratios=[0.5, 1.0, 2.0],
        anchor_strides=[8, 16, 32, 64, 128],
        target_means=[.0, .0, .0, .0],
        target_stds=[1.0, 1.0, 1.0, 1.0]))
# training and testing settings
train_cfg = dict(
    assigner=dict(
        type='MaxIoUAssigner',
        pos_iou_thr=0.5,
        neg_iou_thr=0.4,
        min_pos_iou=0,
        ignore_iof_thr=-1),
    smoothl1_beta=0.11,
    gamma=2.0,
    alpha=0.25,
    allowed_border=-1,
    pos_weight=-1,
    debug=False)
test_cfg = dict(
    nms_pre=1000,
    min_bbox_size=0,
    score_thr=0.05,
    nms=dict(type='nms', iou_thr=0.5),
    max_per_img=100)
# dataset settings
dataset_type = 'CustomDataset'
#dataset_type = 'CocoDataset'
data_root = '/home/ubuntu/MyFiles/mmdetection/data/coco/'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
data = dict(
    imgs_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type=dataset_type,
        ann_file=data_root + 'annotations/input_80.pkl',
        img_prefix=data_root + 'train2017/',
        img_scale=(1333, 800),
        img_norm_cfg=img_norm_cfg,
        size_divisor=32,
        flip_ratio=0.5,
        with_mask=False,
        with_crowd=False,
        with_label=True),
    val=dict(
        type=dataset_type,
        ann_file=data_root + 'annotations/input_20.pkl',
        img_prefix=data_root + 'val2017/',
        img_scale=(1333, 800),
        img_norm_cfg=img_norm_cfg,
        size_divisor=32,
        flip_ratio=0,
        with_mask=False,
        with_crowd=False,
        with_label=True),
    test=dict(
        type=dataset_type,
        ann_file=data_root + 'annotations/input_20.pkl',
        img_prefix=data_root + 'val2017/',
        img_scale=(1333, 800),
        img_norm_cfg=img_norm_cfg,
        size_divisor=32,
        flip_ratio=0,
        with_mask=False,
        with_crowd=False,
        with_label=False,
        test_mode=True))
# optimizer
optimizer = dict(type='SGD',lr=1e-5, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=dict(max_norm=35, norm_type=2))
# learning policy
lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=1.0 / 3,
    step=[8, 11])
checkpoint_config = dict(interval=1)
# yapf:disable
log_config = dict(
    interval=50,
    hooks=[
        dict(type='TextLoggerHook'),
        # dict(type='TensorboardLoggerHook')
    ])
# yapf:enable
# runtime settings
total_epochs = 12
device_ids = range(8)
dist_params = dict(backend='nccl')
log_level = 'INFO'
work_dir = './work_dirs/retinanet_r50_fpn_1x_222'
load_from = None
resume_from = None
workflow = [('train', 1)]

**Comment by youjinqing123** [comment]
We have another question.We have some images without objects that we want to detect,can we also use these images in training?

**Comment by Smoothing97** [comment]
@youjinqing123 sorry but I have the same problem with you.My custom dataset is cutted into 256*256 images,and there are no labels in 80% cutted images.I also have one GTX1080Ti GPU in training step.When I delate all the nolabel cutted training images in my dataset and set :
# optimizer
optimizer = dict(type='SGD', lr=1e-5, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=dict(max_norm=35, norm_type=2))
# learning policy
lr_config = dict(
    policy='step',
    warmup='linear',
   warmup_iters=5000,
    warmup_ratio=1.0 / 3,
    step=[8, 11])
my loss is NAN.
So I really want to know how you solve your problem?thanks!

**Comment by naveenggmu** [comment]
@youjinqing123 How did you solve your issue? What was the problem?
